cmake/std/trilinosprhelpers/setenvironment/ModuleHelper.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python
+"""
+Module helper for *nix systems
+
+This module contains a helper for using the modules subsystem on
+*nix systems.  It will attempt to load the env_modules_python
+module and if that does not exist then we generate our own call
+to the `module()` function.
+
+Todo:
+    Check to find what package provides env_modules_python and document.
+"""
+from __future__ import print_function
+
+import os
+import subprocess
+import sys
+
+if "MODULESHOME" in os.environ.keys():                                             # pragma: no cover
+    sys.path.insert(1, os.path.join(os.environ['MODULESHOME'], 'init'))            # pragma: no cover
+else:                                                                              # pragma: no cover
+    print("WARNING: The environment variable 'MODULESHOME' was not found.")        # pragma: no cover
+    print("         ModuleHelper may not be able to locate modules.")              # pragma: no cover
+
+try:
+
+    from env_modules_python import module
+
+except ImportError:
+
+    def module(*args):
```

Comment:
At some point this routine/file could use more complete documentation.  This is the first thing I've run into in this PR where I'm not really sure what it's doing.

Reply:
This is addressed in the docstring at the top of this file.

I added a comment that explains at the point of the exception, but it's kind of redundant to what is stated at the top of the file.  Essentially, we try and import module() from env_modules_python... if that import fails then we roll our own version of that function.